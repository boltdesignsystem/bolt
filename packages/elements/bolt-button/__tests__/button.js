import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../button.schema';
// eslint-disable-next-line camelcase
const { hierarchy, size, border_radius, display } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: 'This is a button',
    attributes: {
      type: 'button',
    },
  };

  const icon = await render('@bolt-components-icon/icon.twig', {
    name: 'chevron-down',
  });

  fixtures = {
    defaultData,
    icon,
  };
});

describe('Bolt button', () => {
  test(`default`, async () => {
    const results = await render('@bolt-elements-button/button.twig', {
      ...fixtures.defaultData,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt button Props', () => {
  test(`icon after`, async () => {
    const results = await render('@bolt-elements-button/button.twig', {
      ...fixtures.defaultData,
      icon_after: fixtures.icon.html,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  hierarchy.enum.forEach(async option => {
    test(`hierarchy: ${option}`, async () => {
      const results = await render('@bolt-elements-button/button.twig', {
        ...fixtures.defaultData,
        hierarchy: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  size.enum.forEach(async option => {
    test(`size: ${option}`, async () => {
      const results = await render('@bolt-elements-button/button.twig', {
        ...fixtures.defaultData,
        size: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  border_radius.enum.forEach(async option => {
    test(`border_radius: ${option}`, async () => {
      const results = await render('@bolt-elements-button/button.twig', {
        ...fixtures.defaultData,
        border_radius: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  display.enum.forEach(async option => {
    test(`display: ${option}`, async () => {
      const results = await render('@bolt-elements-button/button.twig', {
        ...fixtures.defaultData,
        display: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
});
