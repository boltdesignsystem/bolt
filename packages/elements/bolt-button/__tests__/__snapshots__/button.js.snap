// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt button Props border_radius: full 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--border-radius-full"
>
  This is a button
</button>
`;

exports[`Bolt button Props border_radius: large 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--border-radius-large"
>
  This is a button
</button>
`;

exports[`Bolt button Props border_radius: small 1`] = `
<button type="button"
        class="e-bolt-button"
>
  This is a button
</button>
`;

exports[`Bolt button Props display: block 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--block"
>
  This is a button
</button>
`;

exports[`Bolt button Props display: inline 1`] = `
<button type="button"
        class="e-bolt-button"
>
  This is a button
</button>
`;

exports[`Bolt button Props display: inline@from-small 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--inline@from-small"
>
  This is a button
</button>
`;

exports[`Bolt button Props hierarchy: primary 1`] = `
<button type="button"
        class="e-bolt-button"
>
  This is a button
</button>
`;

exports[`Bolt button Props hierarchy: secondary 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--secondary"
>
  This is a button
</button>
`;

exports[`Bolt button Props hierarchy: tertiary 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--tertiary"
>
  This is a button
</button>
`;

exports[`Bolt button Props hierarchy: transparent 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--transparent"
>
  This is a button
</button>
`;

exports[`Bolt button Props icon after 1`] = `
<button type="button"
        class="e-bolt-button"
>
  This is a button
  <span class="e-bolt-button__icon-after"
        aria-hidden="true"
  >
    <bolt-icon name="chevron-down"
               aria-hidden="true"
    >
      <span class="c-bolt-icon c-bolt-icon--chevron-down">
        <svg class="c-bolt-icon__icon"
             xmlns="http://www.w3.org/2000/svg"
             viewbox="0 0 32 32"
        >
          <path fill="var(--bolt-theme-icon, currentColor)"
                fill-rule="evenodd"
                d="M25 13c.5-.6.5-1.4 0-2-.6-.5-1.4-.5-2 0l-7 7.1-7-7a1.3 1.3 0 00-2.3.9c0 .3.1.7.4 1l8 8c.5.5 1.3.5 1.8 0l8-8z"
          >
          </path>
        </svg>
      </span>
    </bolt-icon>
  </span>
</button>
`;

exports[`Bolt button Props size: large 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--large"
>
  This is a button
</button>
`;

exports[`Bolt button Props size: medium 1`] = `
<button type="button"
        class="e-bolt-button"
>
  This is a button
</button>
`;

exports[`Bolt button Props size: small 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--small"
>
  This is a button
</button>
`;

exports[`Bolt button Props size: xsmall 1`] = `
<button type="button"
        class="e-bolt-button e-bolt-button--xsmall"
>
  This is a button
</button>
`;

exports[`Bolt button default 1`] = `
<button type="button"
        class="e-bolt-button"
>
  This is a button
</button>
`;
